File: src/java/model/Event.java
package model;

import java.sql.Timestamp;
import java.util.Calendar;

public class Event {
    private int id;
    private String name;
    private String description;
    private Timestamp startDate;
    private Timestamp endDate;

    public Event() {
    }

    public int getId() {
        return id;
    }

    public Event setId(int id) {
        this.id = id;
        return this;
    }

    public String getName() {
        return name;
    }

    public Event setName(String name) {
        this.name = name;
        return this;
    }

    public String getDescription() {
        return description;
    }

    public Event setDescription(String description) {
        this.description = description;
        return this;
    }

    public Timestamp getStartDate() {
        return startDate;
    }

    public Calendar getStartDateCalendar() {
        Calendar cal = Calendar.getInstance();
        cal.setTime(startDate);
        return cal;
    }

    public Event setStartDate(Timestamp startDate) {
        this.startDate = startDate;
        return this;
    }

    public Timestamp getEndDate() {
        return endDate;
    }

    public Calendar getEndDateCalendar() {
        Calendar cal = Calendar.getInstance();
        cal.setTime(endDate);
        return cal;
    }

    public Event setEndDate(Timestamp endDate) {
        this.endDate = endDate;
        return this;
    }

    public int getNumberOfSlots() {
        Calendar start = Calendar.getInstance();
        start.setTime(startDate);
        Calendar end = Calendar.getInstance();
        end.setTime(endDate);
        return end.get(Calendar.HOUR_OF_DAY) - start.get(Calendar.HOUR_OF_DAY);
    }
}

File: web/assets/js/dto/Event.js
class Event {
    constructor(id, name, clubId, clubName, startDate, endDate, imgUrl) {
        this.id = id;
        this.name = name;
        this.clubId = clubId;
        this.clubName = clubName;
        this.startDate = startDate;
        this.endDate = endDate;
        this.imgUrl = imgUrl;
    }

    getId() {
        return this.id;
    }

    getName() {
        return this.name;
    }

    getClubName() {
        return this.clubName;
    }

    getStartDate() {
        return this.startDate;
    }

    getEndDate() {
        return this.endDate;
    }

    getImgUrl() {
        return this.imgUrl;
    }
}
